src/main/java/org/testng/internal/MethodHelper.java

```diff
@@ -265,6 +280,33 @@ public int compare(Node<ITestNGMethod> o1, Node<ITestNGMethod> o2) {
     return result;
   }
 
+	/**
+	 * This method is used to create a map of test instances and their associated method(s) .
+	 * Used to decrease the scope to only a methods instance when trying to find method dependencies.
+	 * @param Test Methods to be sorted 
+	 * @return Map of Instances as the keys and the methods associated with the instance as the values
+	 */
+	private static LinkedHashMap<Object, ArrayList<ITestNGMethod>> sortMethodsByInstance(ITestNGMethod[] methods) {

```

Comment:
`Map<Object, List<ITestNGMethod>>` instead of `LinkedHashMap<Object, ArrayList<ITestNGMethod>>`